Mathlib/CategoryTheory/GradedObject.lean

```diff
@@ -238,4 +277,142 @@ end
 
 end GradedObject
 
+namespace GradedObject
+
+variable {I J : Type*} {C : Type*} [Category C]
+  (X Y Z : GradedObject I C) (φ : X ⟶ Y) (e : X ≅ Y) (ψ : Y ⟶ Z) (p : I → J)
+
+/-- If `X : GradedObject I C` and `p : I → J`, `X.mapObjFun p j` is the family of objects `X i`
+for `i : I` such that `p i = j`. -/
+abbrev mapObjFun (j : J) := (fun (i : (p ⁻¹' {j})) => X i)
+
+variable (j : J)
+
+/-- Given `X : GradedObject I C` and `p : I → J`, `X.HasMap p` is the condition that
+for all `j : J`, the coproduct of all `X i` such `p i = j` exists. -/
+abbrev HasMap : Prop := ∀ (j : J), HasCoproduct (X.mapObjFun p j)
+
+variable [X.HasMap p] [Y.HasMap p] [Z.HasMap p]
+
+/-- Given `X : GradedObject I C` and `p : I → J`, `X.mapObj p` is the graded object by `J`
+which in degree `j` consists of the coproduct of the `X i` such that `p i = j`. -/
+noncomputable def mapObj : GradedObject J C := fun j => ∐ (X.mapObjFun p j)
+
+/-- The canonical inclusion `X i ⟶ X.mapObj p j` when `i : I` and `j : J` are such
+that `p i = j`. -/
+noncomputable def ιMapObj (i : I) (j : J) (hij : p i = j) : X i ⟶ X.mapObj p j :=
+  Sigma.ι (fun (i' : (p ⁻¹' {j})) => X i') ⟨i, hij⟩
```

Comment:
I assume this was intended?
```suggestion
  Sigma.ι (mapObjFun X) ⟨i, hij⟩
```

Reply:
Actually, no... Thanks for the suggestions!